From 64256ba78943c52b23ed6fb1638af711b1bb7c2e Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Tue, 13 Aug 2019 01:18:33 +0300
Subject: [PATCH 02/19] net: sh_eth: Toggle PHY GPIO reset pin

Looks like we need to toggle PHY GPIO reset pin.
Otherwise, GEther device fails to start on V3H
platform sometimes.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 drivers/net/sh_eth.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/net/sh_eth.c b/drivers/net/sh_eth.c
index 4cbffb1..eb9d4db 100644
--- a/drivers/net/sh_eth.c
+++ b/drivers/net/sh_eth.c
@@ -761,6 +761,13 @@ static int sh_eth_phy_config(struct udevice *dev)
 	struct phy_device *phydev;
 	int mask = 0xffffffff;
 
+	if (dm_gpio_is_valid(&priv->reset_gpio)) {
+		dm_gpio_set_value(&priv->reset_gpio, 1);
+		mdelay(20);
+		dm_gpio_set_value(&priv->reset_gpio, 0);
+		mdelay(1);
+	}
+
 	phydev = phy_find_by_mask(priv->bus, mask, pdata->phy_interface);
 	if (!phydev)
 		return -ENODEV;
-- 
2.7.4

